numba/cuda/support/pytorch.py

```diff
@@ -0,0 +1,149 @@
+import torch
+import numba.cuda
+
+def tensor_cuda_array_interface(tensor):
+    if not tensor.device.type == "cuda":
+        raise TypeError(
+            "Can't convert cpu tensor to cuda array. "
+            "Use Tensor.cuda() to copy the tensor to device memory first."
+        )
+
+    if tensor.requires_grad:
+        raise RuntimeError(
+            "Can't get cuda array interface for Variable that requires grad. "
+            "Use var.detach() first."
+        )
+
+    if tensor.device.index != numba.cuda.get_current_device().id:
+        raise ValueError(
+            "tensor device: %r is not active numba context: %r" % (
+                tensor.device, numba.cuda.current_contenxt()
+            )
+        )
+
+    typestr = {
+        torch.float16: "f2",
+        torch.float32: "f4",
+        torch.float64: "f8",
+        torch.uint8: "u1",
+        torch.int8: "i1",
+        torch.int16: "i2",
+        torch.int32: "i4",
+        torch.int64: "i8",
+    }[tensor.dtype]
+
+    itemsize = {
+        torch.float16: 2,
+        torch.float32: 4,
+        torch.float64: 8,
+        torch.uint8: 1,
+        torch.int8: 1,
+        torch.int16: 2,
+        torch.int32: 4,
+        torch.int64: 8,
+    }[tensor.dtype]
```

Comment:
Just FYI, one can use `tensor.storage().element_size()` instead of hardcoding this.

Reply:
Great! Thanks for the review. I believe this functionality should be lifted into a separate module to independently track numba & pytorch releases.